src/vivarium_gates_iv_iron/results_processing/process_results.py

```diff
@@ -142,64 +142,78 @@ def aggregate_over_seed(data):
     data = data.join(map_df).reset_index(drop=True)
     data = data.rename(columns=RENAME_COLUMNS)
     logger.info(f"Mapping {kind} complete.")
     return data
 
 
-def get_population_data(data):
-    total_pop = pivot_data(
-        data[
-            [results.TOTAL_POPULATION_COLUMN]
-            + results.RESULT_COLUMNS("population")
-            + GROUPBY_COLUMNS
-        ]
-    )
-    total_pop = total_pop.rename(columns={"key": "measure"})
-    return sort_data(total_pop)
+def get_population_data(data: pd.DataFrame, disaggregate_seeds: bool) -> pd.DataFrame:
+    if disaggregate_seeds:
+        groupby_cols = GROUPBY_COLUMNS + [results.RANDOM_SEED_COLUMN]
+    else:
+        groupby_cols = GROUPBY_COLUMNS
 
+    total_pop = pivot_data(data[[results.TOTAL_POPULATION_COLUMN]
+                                + results.RESULT_COLUMNS('population')
+                                + groupby_cols], disaggregate_seeds)
+    total_pop = total_pop.rename(columns={'key': 'measure'})
+    return sort_data(total_pop, disaggregate_seeds)
 
-def get_measure_data(data, measure):
-    data = pivot_data(data[results.RESULT_COLUMNS(measure) + GROUPBY_COLUMNS])
+
+def get_measure_data(data: pd.DataFrame, measure: str, disaggregate_seeds: bool) -> pd.DataFrame:
+    if disaggregate_seeds:
+        data = pivot_data(
+            data[results.RESULT_COLUMNS(measure) + GROUPBY_COLUMNS + [results.RANDOM_SEED_COLUMN]],
+            disaggregate_seeds
+        )
+    else:
+        data = pivot_data(data[results.RESULT_COLUMNS(measure) + GROUPBY_COLUMNS], disaggregate_seeds)
     data = apply_results_map(data, measure)
-    return sort_data(data)
+    return sort_data(data, disaggregate_seeds)
 
 
-def get_by_cause_measure_data(data, measure):
-    data = get_measure_data(data, measure)
-    return sort_data(data)
+def get_by_cause_measure_data(data: pd.DataFrame, measure: str, disaggregate_seeds: bool) -> pd.DataFrame:
+    data = get_measure_data(data, measure, disaggregate_seeds)
+    return sort_data(data, disaggregate_seeds)
 
 
-def get_state_person_time_measure_data(data, measure):
-    data = get_measure_data(data, measure)
-    return sort_data(data)
+def get_state_person_time_measure_data(data: pd.DataFrame, measure: str, disaggregate_seeds: bool) -> pd.DataFrame:
+    data = get_measure_data(data, measure, disaggregate_seeds)
+    return sort_data(data, disaggregate_seeds)
 
 
-def get_transition_count_measure_data(data, measure):
+def get_transition_count_measure_data(data: pd.DataFrame, measure: str, disaggregate_seeds: bool) -> pd.DataFrame:
     # Oops, edge case.
-    data = data.drop(
-        columns=[c for c in data.columns if "event_count" in c and "2041" in c]
-    )
-    data = get_measure_data(data, measure)
-    return sort_data(data)
+    data = data.drop(columns=[c for c in data.columns if 'event_count' in c and str(results.YEARS[-1]+1) in c])
+    data = get_measure_data(data, measure, disaggregate_seeds)
```

Comment:
Formatting nitpick -- please add spaces around the '+'.

Also, `"event_count"`